File: ws/medina/u2/ws12Inheritance/FarmSystem/src/main/java/ec/edu/espe/farmsystem/model/Sheep.java
package ec.edu.espe.farmsystem.model;

import java.util.Date;

/**
 *
 * @author Joseph B. Medina
 */
public class Sheep extends FarmAnimal {
    
    private Date lastShearing;

    public Sheep(Date lastShearing, int id, String breed, Date bornOn, String gendeR, boolean isAbleToReproduce, float weight, Cage cage, Location location) {
        super(id, breed, bornOn, gendeR, isAbleToReproduce, weight, cage, location);
        this.lastShearing = lastShearing;
    }

    @Override
    public String toString() {
        return "Sheep{" + "lastShearing=" + lastShearing + ", " + super.toString() + '}';
    }
                
    public void cutWhool(float kilogramsOfWool) {
        kilogramsOfWool++;
        
    }
    
    public void shear(String shearedSheep) {
        shearedSheep = "The sheep has been sheared";
        
    }
    
}

File: hw/otavalo/u2/hw19Inheritance/FarmSystemJavaScript/model/Sheep.js
const FarmAnimal = require('./FarmAnimal');

class Sheep extends FarmAnimal {
    constructor(lastShearing, id, breed, bornOn, gendeR, isAbleToReproduce, weight, cage, location) {
        super(id, breed, bornOn, gendeR, isAbleToReproduce, weight, cage, location);
        this.lastShearing = lastShearing;
    }

    toString() {
        return `Sheep{lastShearing=${this.lastShearing}, ${super.toString()}}`;
    }

    cutWhool(kilogramsOfWool) {
        kilogramsOfWool++;
    }

    shear(shearedSheep) {
        shearedSheep = "The sheep has been sheared";
    }
}

module.exports = Sheep;